extensions/ql-vscode/src/helpers.ts

```diff
@@ -266,6 +269,36 @@ export function isFolderAlreadyInWorkspace(folderName: string) {
   );
 }
 
+/** Check if the current workspace is the CodeTour and open the workspace folder.
+ * Without this, we can't run the code tour correctly.
+ **/
+export async function prepareCodeTour(): Promise<void> {
+  if (workspace.workspaceFolders?.length) {
+    const currentFolder = workspace.workspaceFolders[0].uri.fsPath;
+
+    // We need this path to check that the file exists on windows
+    const tutorialWorkspacePath = join(
+      currentFolder,
+      "tutorial.code-workspace",
+    );
+    const toursFolderPath = join(currentFolder, ".tours");
+
+    if (
+      existsSync(tutorialWorkspacePath) &&
+      existsSync(toursFolderPath) &&
+      !isCodespacesTemplate()
+    ) {
+      const tutorialWorkspaceUri = Uri.parse(
+        join(
+          workspace.workspaceFolders[0].uri.fsPath,
+          "tutorial.code-workspace",
+        ),
+      );
+      await commands.executeCommand("vscode.openFolder", tutorialWorkspaceUri);
```

Comment:
Minor: Consider adding some debug logging just before you issue the command, so it's easier to debug this sequence of events in future.

Also consider giving the user a notification message telling them we're about to restart. You can decide whether you want to notify and restart immediately, or whether you want to make this a blocking prompt where you restart only if the user says yes. The spontaneous restart has been one cause of user confusion in workshops when switching to a multiroot workspace file, so we could take the opportunity to reduce such confusion here.

Reply:
Have added some logging and a prompt. This opens immediately when you click on the extension:

<img width="527" alt="Screenshot 2023-03-21 at 11 15 16" src="https://user-images.githubusercontent.com/1354439/226592370-f309c428-7de1-4a6b-b700-1c109b684d18.png">